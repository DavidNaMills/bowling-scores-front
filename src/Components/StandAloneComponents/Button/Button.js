import React from 'react';
import classes from './Button.module.scss';



const Button = ({ click = () => { }, label, isDisabled = false, type = 'default' }) => {
    let style = [classes.button];

    if (isDisabled) {
        style = style.concat(classes.button__disabled);
    } else {

        switch (type.toLowerCase()) {
            case 'warning':
                style = style.concat(classes.button__warning);
                break;
            case 'danger':
                style = style.concat(classes.button__danger);
                break;
            default:
                style = style.concat(classes.button__default);
                break;
        }
    }

    return (
        <button onClick={click} disabled={isDisabled} className={style.join(' ')}>{label}</button>
    );
};

export default Button;