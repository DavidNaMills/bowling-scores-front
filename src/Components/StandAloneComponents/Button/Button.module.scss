@import '../../../styles/colors.scss';


.button{
    border: none;
    font-size: 16px;
    padding: 10px;
    min-width: 100px;
    font-weight: 550;

    &__disabled{
        background-color: $btn_disabled;
    }
    
    &__isFull{
        width: 100%;
    }

    &__default{
        background-color: $btn_default_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_default_bkg_active_hover;
        }
        &:active{
            background-color: $btn_default_bkg_active;
        }
    }

    &__warning{
        background-color: $btn_warning_bkg;
        color: $btn_font_clr;
        
        &:hover{
            background-color: $btn_warning_bkg_active_hover;
        }
        &:active{
            background-color: $btn_warning_bkg_active;
        }    
    }
    
    &__danger{
        background-color: $btn_danger_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_danger_bkg_active_hover;
        }
        &:active{
            background-color: $btn_danger_bkg_active;
        }    
    }

    &__lightGreen{
        background-color: $btn_lightGreen_bkg;
        color: $btn_font_clr;
        
        &:hover{
            background-color: $btn_lightGreen_bkg_active;
        }
        &:active{
            background-color: $btn_lightGreen_bkg_active_hover;
        }    
    }

    &__darkGreen{
        background-color: $btn_darkGreen_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_darkGreen_bkg_active;
        }
        &:active{
            background-color: $btn_darkGreen_bkg_active_hover;
        }    
    }

    &__blue{
        background-color: $btn_blue_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_blue_bkg_active;
        }
        &:active{
            background-color: $btn_blue_bkg_active_hover;
        }    
    }

    &__lightBlue{
        background-color: $btn_lightBlue_bkg;
        color: $btn_font_clr;
        
        &:hover{
            background-color: $btn_lightBlue_bkg_active;
        }
        &:active{
            background-color: $btn_lightBlue_bkg_active_hover;
        }    
    }

    &__darkBlue{
        background-color: $btn_darkBlue_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_darkBlue_bkg_active;
        }
        &:active{
            background-color: $btn_darkBlue_bkg_active_hover;
        }    
    }

    &__lightRed{
        background-color: $btn_lightRed_bkg;
        color: $btn_font_clr;
        
        &:hover{
            color: $btn_font_clr_light;
            background-color: $btn_lightRed_bkg_active;
        }
        &:active{
            background-color: $btn_lightRed_bkg_active_hover;
        }    
    }

    &__lightYellow{
        background-color: $btn_lightYellow_bkg;
        color: $btn_font_clr;
        
        &:hover{
            color: $btn_font_clr_light;
            background-color: $btn_lightYellow_bkg_active;
        }
        &:active{
            background-color: $btn_lightYellow_bkg_active_hover;
        }    
    }



}