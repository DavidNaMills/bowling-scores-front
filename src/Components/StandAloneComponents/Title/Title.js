import React from 'react';
import classes from './Title.module.scss';

const Title = ({ label = '', ttlType = 'sub' }) => {
    let item;

    switch (ttlType.toLowerCase()) {
        case 'main':
            item = <h1 className={classes.title__main}>{label}</h1>
            break;
        case 'section':
            item = <h2 className={classes.title__section}>{label}</h2>
            break;
        case 'sub':
            item = <h3 className={classes.title__sub}>{label}</h3>
            break;
        default:
            item = <h3 className={classes.title__sub}>{label}</h3>
            break;
    }

    return (
        <div className={classes.title}>
            {item}
        </div>
    )
}

export default Title;