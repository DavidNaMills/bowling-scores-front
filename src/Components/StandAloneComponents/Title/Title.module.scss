@import '../../../styles/colors.scss';
@import '../../../styles/fonts.scss';

.title{
    // -webkit-box-shadow: $title_shadow_full;
    // -moz-box-shadow: $title_shadow_full;
    // box-shadow: $title_shadow_full;
    padding: 2px;
    text-align: center;
    font-family: $header;

    &__main{
        font-size: $header_size_main;
        color: $title_main_color;
    }
    
    &__section{
        font-size: $header_size_section;
        color: $title_rest_color;
    }
    &__sub{
        font-size: $header_size_sub;
        color: $title_rest_color;
    }
}