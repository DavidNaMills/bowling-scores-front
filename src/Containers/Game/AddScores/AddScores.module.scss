@import '../../../styles/colors.scss';

.addScores{
    &__form{
        width: 75vw;
        max-width: 400px;
        border: solid $popup_color 2px;
    }
}