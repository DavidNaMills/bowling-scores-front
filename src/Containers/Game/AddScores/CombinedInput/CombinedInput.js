import React from 'react';
import classes from './CombinedInput.module.scss';

import Input from '../../../../Components/Form/elements/Input/Input';
import { playerTableStyle } from '../../playerTableStyle';

//Avatar when built

const CombinedInput = ({ name, color, value, updateScore, id }) => {

    // <div style={playerTableStyle()}>
    return (
        <div style={playerTableStyle(color)} className={classes.combinedInput}>
            <div className={[classes.combinedInput__name, classes.combinedInput].join(' ')}>
                <p>{name}</p>
            </div>

            <div className={[classes.combinedInput__input, classes.combinedInput].join(' ')}>
                <Input
                    value={value}
                    changed={e => updateScore({ id, data: e.target.value })}
                    id={id}
                    type='number'
                    max={300}
                    min={0}
                />
            </div>
        </div>
    )
}

export default CombinedInput;