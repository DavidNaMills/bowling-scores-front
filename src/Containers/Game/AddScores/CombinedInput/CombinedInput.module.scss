.combinedInput{
    display: flex;
    justify-content: space-around;
    align-items: center;
    // padding: 10px;

    &__name{
        width: 60%;
        padding: 0;
        margin: 0;
    }
    
    &__input{
        justify-content: left;
        width: 40%;
    }
}