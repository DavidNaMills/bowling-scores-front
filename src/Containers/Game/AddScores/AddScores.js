import React, { useEffect } from 'react';
import CombinedInput from './CombinedInput/CombinedInput';
import objectToArray from '../../../helpers/objectToArray/objectToArray'
import Button from '../../../Components/StandAloneComponents/Button/Button';
import Title from '../../../Components/StandAloneComponents/Title/Title';

import useScoreInput from '../../../Hooks/useScoreInput/useScoreInput';
import useDispatchHook from '../../../Hooks/useDispatchHook/useDispatchHook';

const AddScores = ({ players, close }) => {
    const { values, addValue } = useScoreInput(players);
    const {addNewGameDispatch} = useDispatchHook();

    const updateScore = (e) => {
        e.preventDefault();
        let touched = false;

        for (let k in values) {
            if (values[k].score > 0) touched = true;
        }
        if (touched) {
            addNewGameDispatch(values);
            console.log(values);
            close();
        } else {
            alert('must add at least 1 score')
        }
    }

    useEffect(() => { }, [values]);

    return (
        <div>
            <Title label='Add Scores' ttlType='sub' />
            {
                values ?
                    <React.Fragment>
                        <form onSubmit={updateScore}>
                            {
                                objectToArray(players).map((x, i) =>
                                    <CombinedInput
                                        key={i}
                                        name={x.config.name}
                                        color={x.config.color}
                                        updateScore={addValue}
                                        id={x.id}
                                        value={values[x.id].score}
                                    />
                                )}
                            <Button click={() => { }} isFull label='Save Scores' />
                        </form>

                        <div>
                            <Button isFull click={close} label='Close' />
                        </div>
                    </React.Fragment>
                    : <p>Please wait</p>
            }
        </div>
    )
}

export default AddScores;