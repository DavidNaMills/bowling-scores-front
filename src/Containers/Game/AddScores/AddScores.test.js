import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';

import AddScores from './AddScores';
import CombinedInput from './CombinedInput/CombinedInput';
import Button from '../../../Components/StandAloneComponents/Button/Button';
import Title from '../../../Components/StandAloneComponents/Title/Title';


const mockPlayers = {
    '1ad602da-92de-417e-86f6-3cd1d84bc328': {
        name: "david",
        color: "233, 80, 237"
    },
    '1ad602da-92de-86f6-3cd1d84bc328': {
        name: "alan",
        color: "233, 20, 237"
    },
    '92de-417e-86f6-3cd1d84bc328': {
        name: "bob",
        color: "233, 180, 237"
    },
}

const close = jest.fn();


describe('<AddScores/> test suite', () => {
    it('placeholder', ()=>{
        expect(true).toBeTruthy();
    });

    // describe('snapshot tests', () => {
    //     const component = renderer.create(<AddScores players={mockPlayers} close={close} />)
    //     const app = component.toJSON();
    //     expect(app).toMatchSnapshot();
    // });

    // describe('render test suite', () => {
    //     const wrapper = mount(<AddScores players={mockPlayers} close={close} />);
        
    //     it('renders a <Title>', ()=>{
    //         expect(wrapper.find(Title).length).toBe(1);
    //     });

    //     it('renders a form with accompanying containers', ()=>{
    //         expect(wrapper.find('.form__container').length).toBe(1);
    //         expect(wrapper.find('.form__form').length).toBe(1);
    //         expect(wrapper.find('form').length).toBe(1);
    //     });
        
    //     it(`form contains ${Object.keys(mockPlayers).length} <CombinedInputs`, ()=>{
    //         expect(wrapper.find(CombinedInput).length).toBe(Object.keys(mockPlayers).length);
    //     });

    //     it('contains 2 <Buttons /> with accompanying containers', ()=>{
    //         expect(wrapper.find('.largeExtra').length).toBe(2);
    //         expect(wrapper.find(Button).length).toBe(2);
    //     });
    // });

});