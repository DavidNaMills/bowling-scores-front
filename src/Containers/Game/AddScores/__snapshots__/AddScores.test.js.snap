// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="contentContainer"
>
  <div
    className="title"
  >
    <h3
      className="title__sub"
    >
      Add Scores
    </h3>
  </div>
  <p>
    Please wait
  </p>
</div>
`;
